### PR TITLE
improve feature: make list item clickable in touch device

### PR DESCRIPTION
**Pull request by wewea**
### make the table clickable and add a example for dev test

issue: https://github.com/weflex/react-touchable-tabular/issues/1

refer to the implement in admin

run example:  run `npm install && npm run dev` inside `example`, open the link `http://localhost:8080` and you can test touch in chrome devtool Device Mode. i have add clickEventListener on row1, so you can click any cell in row1 and it will alert an messenger.

BTW, npm@3 had change the behavior of peerDependecies that [stop auto-installing peerDependencies](https://github.com/npm/npm/issues/6565) and if we want to run the example we have to install moment, react, and react-dom  manually, should i add these library to devDenpencies?

R=@yorkie

### CODE REVIEW

**Comment by yorkie** [comment]
Hey @wewea remember using `R={someone}` to take a reviewer for any PR under https://github.com/weflex.

File: example/example.js [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+          'last': 'kanshan',
+        },
+      },
+      {
+        'name': {
+          'first':'zhihu',
+          'last': 'kanshan',
+        },
+      },
+    ]
```
Comment (yorkie) [inline example/example.js]: More the array to separate JSON file?

File: example/example.js [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+  }
+  render() {
+    return (
+      <SortedTable getDataSource={this.getDataSource}
+      tableHeight={1000}
```
Comment (yorkie) [inline example/example.js]: miss indentions here

File: package.json [hunk 3]
```diff
@@ -25,7 +25,7 @@
   "homepage": "https://github.com/weflex/react-touchable-tabular#readme",
   "dependencies": {
     "fixed-data-table": "0.6.0",
-    "scroller": "0.0.3"
+    "hammerjs": "^2.0.6"
```
Comment (yorkie) [inline package.json]: use exact version here

File: src/index.jsx [hunk 4]
```diff
@@ -3,16 +3,14 @@ require('fixed-data-table/dist/fixed-data-table.min.css');
...
 const {
-  Scroller
-} = require('scroller');
-const {
   Table,
   Column,
   Cell
 } = require('fixed-data-table');
-
+const Hammer = require('hammerjs');
```
Comment (yorkie) [inline src/index.jsx]: Could we use `import` here as well?

File: src/index.jsx [hunk 5]
```diff
@@ -20,115 +18,10 @@ var SortTypes = {
 };
 
 function isTouchDevice() {
-  return 'ontouchstart' in document.documentElement // works on most browsers
+  return 'ontouchstart' in document// works on most browsers
```
Comment (yorkie) [inline src/index.jsx:17]: Why removing `documentElement`?

File: src/index.jsx [hunk 6]
```diff
@@ -220,8 +112,38 @@ class Tabular extends React.Component {
...
+      } else {
+        this.setState({[topProps]: scrollTop});
+      }
+
+      // TODO(jkvim) implement scroll horizontally in the future
+      // var scrollLeft = this.state.left - ev.deltaX;
+      // if (scrollLeft <= 0) {
+      //   this.setState({[leftProps]: 0});
+      // } else if (scrollLeft >= this.state.maxScrollY) {
+
```
Comment (yorkie) [inline src/index.jsx]: remove this line

File: src/index.jsx [hunk 7]
```diff
@@ -299,26 +231,44 @@ class Tabular extends React.Component {
...
+        overflowX: 'hidden',
+        scrollTop: this.state.scrollTop,
+        scrollLeft: this.state.scrollLeft,
+      });
+    }
   }
 }
 
-module.exports = Tabular;
+module.exports = SortedTable;
```
Comment (yorkie) [inline src/index.jsx]: revert this line, we should name it to be `Tabular`

**Comment by yorkie** [comment]
@jkvim Round 1 is done, left some comments.

File: src/index.jsx (hunk 5 above)
Comment (wewea) [inline src/index.jsx:17]: @yorkie documentElement haven't ontouch\* event

File: src/index.jsx (hunk 6 above)
Comment (wewea) [inline src/index.jsx]: @yorkie you meant all comment line or the line 140

File: src/index.jsx (hunk 5 above)
Comment (yorkie) [inline src/index.jsx:17]: We should check both all cases

File: src/index.jsx (hunk 6 above)
Comment (yorkie) [inline src/index.jsx]: yup

**Comment by wewea** [comment]
@yorkie  update again, change file to json, please review.

**Comment by yorkie** [comment]
Thank you, awesome work :-)

**Pull request merged**